downstairs/src/region.rs

```diff
@@ -605,11 +649,46 @@ impl Region {
             }
         };
         save_stream_to_file(extent_copy, repair_stream.into_inner()).await?;
+        let mut count = 1;
+
+        // If we are replacing our region with one from an older version
+        // that contained the SQLite, then look for and copy the database
+        // files.
+        for opt_file in &[ExtentType::Db, ExtentType::DbShm, ExtentType::DbWal]
```

Comment:
I believe we could do a loop over `[ExtentType::Data, ExtentType::Db, ExtentType::DbShm, ExtentType::DbWal]` and remove the specific handling of the data file above (L635-651).

Reply:
Yes, with the validate_*_files()` function, we know we have a correct set of files to stream over.

Reply:
Done in [7bef555](https://github.com/oxidecomputer/crucible/commit/7bef55560bd1a345855ca77b6c296616ee66e0fe)